### ISSUE TITLE
SVG of the marker tool is not rendered properly

### ISSUE THREAD

**Issue by mmahmoudian** [flameshot-org/flameshot#2667]
### Flameshot Version

Flameshot v12.0.0 (6f6a8dde)

### Installation Type

Compiled from source

### Operating System type and version

Fedora 36 KDE wayland

### Description

When compiled from source, the against 6f6a8dde commit, the marker tool's icons is not rendered properly. This is not the case when installed form AUR (same commit) on Manjaro KDE X11.

### Steps to reproduce

```sh
dnf install flameshot                     # Hack to install runtime files
dnf install kf5-kguiaddons-devel          # Required for wayland clipboard on KDE
dnf install qt5-linguist qt5-qtsvg-devel  # these were not installed by the above hack and were not pre-installed in Fedora KDE spin


git clone https://github.com/flameshot-org/flameshot
cd flameshot
mkdir build
cd build
cmake ../


./src/flameshot gui
```

### Screenshots or screen recordings

![](https://user-images.githubusercontent.com/390889/173604468-22e19f3d-754b-4411-8bcc-349e28353ec6.png)

### System Information

Virtual Machine Manager v4.0.0:

![image](https://user-images.githubusercontent.com/390889/173609007-58db40a0-ee31-43cc-a040-9dac6f73cedd.png)

**Comment by borgmanJeremy** [comment]
@RiedleroD Are you able to help on this? It's not urgent I can roll back just that one file until you have time to get to it. I also replicated it in a Fedora KDE VM.

**Comment by RiedleroD** [comment]
looks fine to me. how is it supposed to look?

**Comment by borgmanJeremy** [comment]
@RiedleroD  look at the purpose icon not the orange.

**Comment by mmahmoudian** [comment]
> looks fine to me. how is it supposed to look?

The orange is what we used to have and aiming to have and the purple is the one we currently have. The lower part seems to be missing. I'm no svg expert and have no idea why on some computers it looks ok and why on others it doesn't

**Comment by RiedleroD** [comment]
ohh! sure, I'll have a look at what I messed up last time ;P

tomorrow though. It's become later than expected today.

**Comment by mmahmoudian** [comment]
@RiedleroD thanks for your time and contributions 👍

**Comment by RiedleroD** [comment]
I can't reproduce, even rsvg renders this correctly. I'm assuming this is a temporary bug in Qt's svg rendering, but I'll try to look further into it

**Comment by RiedleroD** [comment]
also @mmahmoudian your steps to reproduce lack the make command, fyi

**Comment by RiedleroD** [comment]
my final opinion: this is either a problem with how Qt renders SVGs that is fixed in a newer version, which arch-based distros already have, and fedora doesn't. Try finding out if you can temporarily install a newer version of qt5-qtsvg-devel on your fedora VM.

Also, I've said it, and I'll say it again: Qt's svg renderer sucks and we should switch to resvg or something if possible.

**Comment by mmahmoudian** [comment]
> also @mmahmoudian your steps to reproduce lack the make command, fyi

Yeah, you are right, sorry about that. I did itin a VM and my clipboard was not mapped to copy exactly the sequence of commands from there. I typed some of it and copied the rest from Jeremy's comment in another issue.

**Comment by RiedleroD** [comment]
> I have the VM. I can give you a download link if you want

please do. after you've tried the thing below ↓

> I can try to create a manjaro Wayland KDE VM and see if it is reproducible there as well or not.

that would be nice. If the same problem persists, try fedora X11 and something else than KDE as well please.

**Comment by mmahmoudian** [comment]
This issue is not reproducible in Manjaro KDE Wayland:

![image](https://user-images.githubusercontent.com/390889/173816957-084f3a19-8846-4e1d-afc3-058e502c8fa1.png)


@borgmanJeremy So it seems @RiedleroD hypothesis is correct. But considering that we have a huge Ubuntu and Fedora userbase (my assumption based on the bug reports), I suggest the following in this order

1. if possible and feasible, switch to the resvg or whatever seems appropriate as per suggestion of @RiedleroD 
2. if previous is not possible, for the time being we revert the optimization for this particular file and perhaps merge it back in the repo after half/one year to cause less issues for users

Of course I have no solid understanding of the work the first suggestion needs, so take it lightly considering that we are getting prepared for v12

**Comment by RiedleroD** [comment]
I've got no idea how easy this is in practice, but resvg has a [drop-in for QtSVG](https://github.com/RazrFalcon/resvg/blob/master/c-api/ResvgQt.h).

I can't write cmake files nor C++ code, otherwise I'd open a PR for this.

**Comment by borgmanJeremy** [comment]
I tested the RPM built by #2668  and its working on Fedora 36
